### ISSUE TITLE
Increase Nmesh?

### ISSUE THREAD

**Issue by sbird** [MP-Gadget/MP-Gadget#279]
I have been investigating increasing the default Nmesh from 2x npart^{1/3} to 3x npart ^{1/3}.

This has a negative effect on both simulation speed and memory usage. For a 1Gpc/h 2x768^3 simulation with no hydro or feedback, I measured the PM taking 3.5 times (3**3/2**3) longer. This was partially compensated because the tree gravity walk took was 1.5 times slower. 

The total performance hit was a decrease of about ~10 %  (because the tree build is a lot of the tree time, so the faster tree gravity doesn't count). PM went from ~10% of runtime to ~25% of runtime. It is probably less at higher resolution and with more nonlinearities. The memory cost is about 20%: with 2xnpart the peak memory usage is with density, with 3xnpart it is PM again.

So why would we do this? This is a power spectrum plot:

![Nmesh2v3](https://user-images.githubusercontent.com/1325948/57002843-31112c00-6b77-11e9-8f7a-953b62f61a6a.png)

The large-scale effect of the increased mesh is very close to zero (ignore that CLASS doesn't agree with the linear power: it is because the redshifts are slightly different). The increased Nmesh seems to noticeably damp down systematics near the particle grid scale, I think because with 3xnpart the PM mesh is no longer a multiple of the particle grid. The oscillatory beating effects are gone much earlier. Of course we don't get a converged power spectrum on the particle grid scale anyway (at least not until non-linear growth really kicks in), but it might help us within halos.

Yu, what do you think? Worth it or not?

**Comment by rainwoodman** [comment]
I think this is just pushing the PM-split problem to a smaller scale. So regardless of B=3 or B=2 we still need to fix the PM-split by enlarging r_split to 1.5 or even 2 from 1.25. By going to B=3 it means we can increase r_split to 1.25 * 1.5 = 1.875 for free.

So I'd say go for it, and perhaps recalibrate to a larger r_split together while you are at it.

**Comment by sbird** [comment]
Ah yes! That is a good point which I had not thought of. I will change the default and start thinking about recalibrating r_split in a separate PR.

**Comment by rainwoodman** [comment]
Sure! You may want to experiment with the erfc short range first, because then you don't need to rerun the kernel and change a bunch of code for each experiment.

**Comment by sbird** [comment]
A good idea!

**Comment by rainwoodman** [comment]
How does the memory usage go? I wonder if memory is still OK going to B=4. If memory is OK and if we can drop number of PM steps by a factor of 4 or 5, then ...

**Comment by sbird** [comment]
It gets large fast. B=4 is probably too much memory: it goes like B**3

**Comment by sbird** [comment]
Implemented in #350
